chat/src/main/java/com/github/twitch4j/chat/TwitchChatConnectionPool.java

```diff
@@ -255,6 +256,9 @@ protected TwitchChat createConnection() {
                 .withAutoJoinOwnChannel(false) // user will have to manually send a subscribe call to enable whispers. this avoids duplicating whisper events
         ).build();
 
+        // Reclaim channel headroom upon generic join failures
+        chat.getEventManager().onEvent(threadPrefix + "join-fail-tracker", ChannelRemovedPostJoinFailureEvent.class, e -> unsubscribe(e.getChannelName()));
```

Comment:
Should we take `automaticallyPartOnBan` to mean `removeChannelOnJoinFailure` (technically results in parts in more scenarios than just ban)

Or, should we deprecate `automaticallyPartOnBan` in favor of `removeChannelOnJoinFailure`

Reply:
I think we can deprecate `automaticallyPartOnBan` in favor of `removeChannelOnJoinFailure`. The reason on why the join failed isn't really important imo, either we want to rejoin or we don't.